### DISCUSSION TITLE
Cannot start a runtime from within a runtime

### DISCUSSION THREAD

**Discussion by Virgil-N** [tokio-rs/tokio#5274] [Q&A] [answered]
**Version**
1.22.0

**Platform**
Mac M1

**Description**
Enter your issue details here.
One way to structure the description:

I tried this code:
```
let rt1 = tokio::runtime::Runtime::new().unwrap();
rt1.block_on(async {
    self.render_news_cards(ui).await;
});
```
start with cargo run, error occurs:
thread 'main' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.'

How to fix it?

**Comment by Darksonn** [comment, marked as answer]
Your code example is too small to show what the issue is. Did you write that code inside a `#[tokio::main]` function? If you're using `block_on`, then you shouldn't use `#[tokio::main]`.

**Reply by Virgil-N** [reply]
Yes, I had used the attr. Should I remove it?

**Reply by Darksonn** [reply]
The `#[tokio::main]` attr creates a runtime, which you should only do once.

**Reply by Virgil-N** [reply]
Can't nest?

**Reply by Darksonn** [reply]
No, runtimes cannot nest.

**Reply by Virgil-N** [reply]
Got it.

**Reply by ShyunnY** [reply]
@Darksonn 

I have a problem: the third-party crate I introduced manually builds the async Runtime through `tokio::runtime::Builder`. At this time, I need to reference this crate in my project.

Based on your discussion, can I understand that when I introduce this crate, I cannot add `[tokio::main]` to `fn main()`, otherwise it will not work properly.

Is my understanding correct?
